cmd/deploy.go

```diff
@@ -472,3 +490,25 @@ func validateBuildType(buildType string) error {
 	}
 	return nil
 }
+
+func parseImageDigest(imageSplit []string, config deployConfig) error {
+
+	if !strings.HasPrefix(imageSplit[1], "sha256:") {
+		return fmt.Errorf("value '%s' in --image has invalid prefix syntax for digest (should be 'sha256:')", config.Image)
+	}
+
+	if len(imageSplit[1][7:]) != 64 {
+		return fmt.Errorf("sha256 hash in '%s' from --image has the wrong length (%d), should be 64", imageSplit[1], len(imageSplit[1][7:]))
+	}
+
+	// if BuidType was not explicitly set via CLI, set to 'disabled' without an error
+	if config.BuildType != "" && config.BuildType != "disabled" {
+		return fmt.Errorf("build type '%s' is not accepted with --image with digest. Use 'disabled' or none", config.BuildType)
+	}
+
+	if config.Push {
+		return fmt.Errorf("--image was specified with digest, therefore --push is not allowed")
```

Comment:
```suggestion
		return fmt.Errorf("The --deploy flag is not valid when using --image with a digest")
```
To be consistent with the second suggestion above for the `--build` flag.

Reply:
i realized the --push flag is true by default and therefore it has to be specified as `--push=false` in order to work with digests so i thought of:
1. change error message to something more explicit like: "the --push flag has to be false when using --image with a digest"
2. set --push to false explicitly in the code and print out a warning if left as true (by default) or by a user via CLI

(following point 2 i could explicitly set `--build` to disabled and print out a warning if it was not disabled)

would any of these suggestions be good or should i keep your initial suggestion?

Reply:
My personal preference is number 2, setting `--push` to false in the code and printing a warning. I think `--build` could possibly work the same way. But we must tread carefully. Explicitly setting `--push` and `--build` configuration to essentially noop should probably only happen if the user has not specified anything at all, and instead these values are just the defaults. If the user has explicitly set the `--build` or `--push` flag to something invalid, and ALSO has an `--image` tag with a digest, we can't really know the intent, so perhaps failing fast in this case would be better.